core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala

```diff
@@ -387,6 +387,36 @@ class BitcoinTxBuilderTest extends BitcoinSAsyncTest {
     }
   }
 
+  it must "succeed to sign a cltv spk that uses a second-based locktime" in {
+    val fundingPrivKey = ECPrivateKey.freshPrivateKey
+
+    val cltvSPK =
+      CLTVScriptPubKey(ScriptNumber(System.currentTimeMillis / 1000),
+                       P2PKScriptPubKey(fundingPrivKey.publicKey))
+
+    val cltvSpendingInfo = LockTimeSpendingInfo(
+      TransactionOutPoint(DoubleSha256DigestBE.empty, UInt32.zero),
+      Bitcoins.one,
+      cltvSPK,
+      Vector(fundingPrivKey),
+      HashType.sigHashAll,
+      ConditionalPath.NoConditionsLeft
+    )
+
+    val txBuilderF =
+      BitcoinTxBuilder(
+        Vector(
+          TransactionOutput(Bitcoins.one - CurrencyUnits.oneMBTC,
+                            EmptyScriptPubKey)),
+        Vector(cltvSpendingInfo),
+        SatoshisPerByte(Satoshis.one),
+        EmptyScriptPubKey,
+        RegTest
+      )
+
+    txBuilderF.flatMap(_.sign).map(_ => succeed)
```

Comment:
Shouldn't we verify this in the script interpreter?

Reply:
That shouldn't be necessary as we are only testing `TxBuilder.calcLockTime` here. That said, I did change this to verify that the transaction's `lockTime` is correct rather than just `suceed`ing